parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/ConsumerManager.java

```diff
@@ -55,7 +67,36 @@ ConsumerRecords<K, V> poll(Duration requestedLongPollTimeout) {
             pollingBroker.set(true);
             updateCache();
             log.debug("Poll starting with timeout: {}", timeoutToUse);
-            records = consumer.poll(timeoutToUse);
+            Instant pollStarted = Instant.now();
+            long tryCount = 0;
+            try {
+                pendingRequests.addAndGet(1L);
+                while (!shutdownRequested.get()) {
+                    tryCount++;
+                    try {
+                        records = consumer.poll(timeoutToUse);
+                        break;
+                    } catch (SaslAuthenticationException authenticationException) {
+                        Instant now = Instant.now();
+                        Duration elapsed = Duration.between(pollStarted, now);
+                        boolean shouldRetry = elapsed.toMillis() < saslAuthenticationRetryTimeout.toMillis();
+                        if (shouldRetry) {
+                            log.warn("Poll error: SaslAuthenticationException. Retrying ({})", tryCount);
+                            try {
+                                retryBackOff(5000L); // no need to check return value here as next loop will check
```

Comment:
can you please change this to constant and reduce to 1 second - so that timeout can be set a bit more granular.

Reply:
Will make it a constant for retry backoff, but the retryBackOff is already checking the shutdownRequested flag in 100ms intervals. It will return early so the while loop will break by itself as early as possible.